irispy/new_sji.py

```diff
@@ -222,62 +291,66 @@ def read_iris_sji_level2_fits(filename, memmap=False):
+            unit = iris_tools.DN_UNIT["SJI_UNSCALED"]
+            uncertainty = None
+        elif not memmap:
+            data_nan_masked[data == BAD_PIXEL_VALUE_SCALED] = np.nan
+            mask = data_nan_masked == BAD_PIXEL_VALUE_SCALED
+            scaled = True
+            # Derive unit and readout noise from the detector
+            unit = iris_tools.DN_UNIT["SJI"]
+            readout_noise = iris_tools.READOUT_NOISE["SJI"]
+            # Derive uncertainty of data for NDCube from fits file.
+            uncertainty = u.Quantity(np.sqrt((data_nan_masked*unit).to(u.photon).value
+                                             + readout_noise.to(u.photon).value**2),
+                                     unit=u.photon).to(unit).value
+        # Derive exposure time from detector.
+        exposure_times = hdulist[1].data[:, hdulist[1].header["EXPTIMES"]]
+        # Derive extra coordinates for NDCube from fits file.
+        times = np.array([parse_time(hdulist[0].header["STARTOBS"])
+                          + timedelta(seconds=s)
+                          for s in hdulist[1].data[:, hdulist[1].header["TIME"]]])
+        pztx = hdulist[1].data[:, hdulist[1].header["PZTX"]] * u.arcsec
+        pzty = hdulist[1].data[:, hdulist[1].header["PZTY"]] * u.arcsec
+        xcenix = hdulist[1].data[:, hdulist[1].header["XCENIX"]] * u.arcsec
+        ycenix = hdulist[1].data[:, hdulist[1].header["YCENIX"]] * u.arcsec
+        obs_vrix = hdulist[1].data[:, hdulist[1].header["OBS_VRIX"]] * u.m/u.s
+        ophaseix = hdulist[1].data[:, hdulist[1].header["OPHASEIX"]]
+        extra_coords = [('TIME', 0, times), ("PZTX", 0, pztx), ("PZTY", 0, pzty),
+                        ("XCENIX", 0, xcenix), ("YCENIX", 0, ycenix),
+                        ("OBS_VRIX", 0, obs_vrix), ("OPHASEIX", 0, ophaseix),
+                        ("EXPOSURE TIME", 0, exposure_times)]
+        # Extraction of meta for NDCube from fits file.
+        startobs = hdulist[0].header.get('STARTOBS', None)
+        startobs = parse_time(startobs) if startobs else None
+        endobs = hdulist[0].header.get('ENDOBS', None)
+        endobs = parse_time(endobs) if endobs else None
+        meta = {'TELESCOP': hdulist[0].header.get('TELESCOP', None),
+                'INSTRUME': hdulist[0].header.get('INSTRUME', None),
+                'TWAVE1': hdulist[0].header.get('TWAVE1', None),
+                'STARTOBS': startobs,
+                'ENDOBS': endobs,
+                'NBFRAMES': hdulist[0].data.shape[0],
+                'OBSID': hdulist[0].header.get('OBSID', None),
+                'OBS_DESC': hdulist[0].header.get('OBS_DESC', None)}
+        list_of_cubes.append(IRISMapCube(data_nan_masked, wcs, uncertainty=uncertainty,
+                                         unit=unit, meta=meta, mask=mask,
+                                         extra_coords=extra_coords, scaled=scaled))
+        hdulist.close()
+
+    return IRISMapCubeSequence(list_of_cubes, meta=meta, common_axis=0)
```

Comment:
I think an ```IRISMapCubeSequence``` should only be returned if there is more than one file input.  Otherwise, an ```IRISMapCube``` should be returned as before.

Reply:
Ok, this is boring to use it with only one file (I am working with). I will modify it.